rmb-sdk-go/peer/peer.go

```diff
@@ -147,13 +162,58 @@ func NewPeer(
 		reader:  reader,
 		writer:  writer,
 		handler: handler,
+		encoder: e,
 	}
 
 	go cl.process(ctx)
 
 	return cl, nil
 }
 
+type peerCfg struct {
+	relayURL         string
+	keyType          string
+	session          string
+	enableEncryption bool
+	schema           encoder.Schema
+}
+
+type PeerOpt func(*peerCfg)
+
+func WithSession(session string) PeerOpt {
+	return func(p *peerCfg) {
+		p.session = session
+	}
+}
+
+func WithEncryption(enable bool) PeerOpt {
+	return func(p *peerCfg) {
+		p.enableEncryption = enable
+	}
+}
+
+func WithRelay(url string) PeerOpt {
+	return func(p *peerCfg) {
+		p.relayURL = url
+	}
+}
+
+func WithKeyType(keyType string) PeerOpt {
+	return func(p *peerCfg) {
+		// to ensure only ed25519 and sr25519 are used
+		if keyType != KeyTypeEd25519 {
+			keyType = KeyTypeSr25519
+		}
+		p.keyType = keyType
+	}
+}
+
+func WithEncoder(schema encoder.Schema) PeerOpt {
```

Comment:
Check the other comment, this instead should take `Encoder` not Schema! to make it easier for developer to implement their own encoder types. without the need to modify this library code